src/parser/Parser.ts

```diff
@@ -2830,6 +2856,22 @@ export class Parser {
         return this.tokens[this.current - 1];
     }
 
+    /**
+     * Get the token that is {diff} indexes away from {this.current}
+     * @param diff the number of steps away from current index to fetch
+     * @param tokenKinds the desired token must match one of these
+     * @example
+     * getToken(-1); //returns the previous token.
+     * getToken(0);  //returns current token.
+     * getToken(1);  //returns next token
+     */
+    private getToken(diff: number, ...tokenKinds: TokenKind[]): Token {
```

Comment:
Maybe should be more explicit like `getPreviousTokenMatchingAny`

Reply:
Good point. I renamed it `getMatchingTokenAtOffset`